### PR TITLE
Lidar oe edit

### PR DESCRIPTION
**Pull request by SamanthaDawsonBanks**

### CODE REVIEW

File: mcomp_C/common/datatypes/Waypoint.cpp [hunk 1]
```diff
@@ -25,8 +25,8 @@ Waypoint::Waypoint(double xIn, double yIn) {
 Waypoint::Waypoint(AngleDistance a) {
   // TODO Auto-generated constructor stub
   a.getDistance();
-  y = a.getDistance() * sin(a.getTheta());
-  x = a.getDistance() * cos(a.getTheta());
+  y = a.getDistance() * sin(a.getTheta()* (M_PI / 180));
```
Comment (SamanthaDawsonBanks) [inline mcomp_C/common/datatypes/Waypoint.cpp:28]: if we don't tell Steve he might not notice

File: mcomp_C/lsensor/LSensor.cpp [hunk 2]
```diff
@@ -72,96 +234,113 @@ unsigned int LSensor::getAvgRPM(){
...
 
-unsigned int* LSensor::decodeRead(){
+unsigned int* LSensor::decodeRead() {
   int counter = 0;
-  for(int i = 0; i < 90; i = i + 4){
-    counter += 4;
-    for(int j = 0;  j < 4; j++){
-      distances[i*4+j] = getRead(counter);
-      counter += 4;
+  for (int i = 0; i < 90; i = i + 1) {
```
Comment (SamanthaDawsonBanks) [inline mcomp_C/lsensor/LSensor.cpp]: the main packet issue

**Comment by SamanthaDawsonBanks** [comment]
Also we probably need to edit the return 0 for bad data as that places a wp (and thus vertex) at the robot's location

**Comment by SamanthaDawsonBanks** [comment]
I think this is as far as we can take the ode without testing with another LiDAR Sensor to rule out bad hardware.
The test data passes the code fine, but the RPM and distance flag up about 99% errors

File: mcomp_C/common/datatypes/Waypoint.cpp (hunk 1 above)
Comment (stevempope) [reply mcomp_C/common/datatypes/Waypoint.cpp:28]: Ha, ha, ha

**Review by stevempope — APPROVED** [review]

File: mcomp_C/communication/Pipe.cpp [hunk 3]
```diff
@@ -14,12 +14,15 @@
 
 #include "Pipe.h"
 
-LSensor ls;
-Propulsion p;
+#include "../config/robot_config.h"
```
Comment (RyanShoobert) [inline mcomp_C/communication/Pipe.cpp:17]: Moving the default baud rate to the config file makes much more sense, good spot

**Review by RyanShoobert — APPROVED** [review]

**Pull request merged**